src/command/check/common.c

```diff
@@ -59,7 +59,9 @@ checkDbConfig(const unsigned int pgVersion, const unsigned int pgIdx, const Db *
         const String *dbPath = dbPgDataPath(dbObject);
 
         // Error if the version from the control file and the configured pg-path do not match the values obtained from the database
-        if (pgVersion != dbVersion || strCmp(cfgOptionIdxStr(cfgOptPgPath, pgIdx), dbPath) != 0)
+        // Skip the db version check if the pg-option is provided since we want to force that value
+        if ((pgVersion != dbVersion && !cfgOptionTest(cfgOptPgVersion)) ||
+            strCmp(cfgOptionIdxStr(cfgOptPgPath, pgIdx), dbPath) != 0)
```

Comment:
It is not clear to me why we need this check. If the version was already forced then it should match, right?

Reply:
Well, the `pgVersion != dbVersion` check was there before the pg-version option and checks if `server_version_num` matches the `pgInterface[].version`.

What we are doing with pg-version is forcing the `pgInterface[].version`. But the queried `server_version_num` might still not match (I don't know how other forks work with that number but it could be changed to match the fork version and not the PG version itself). We are not forcing 

I believe that check is important when we don't use the pg-version option. The skip part is there because we are not forcing `dbVersion`. If we did, then pgVersion and dbVersion would match indeed.

Reply:
Hmmm, I'm not convinced. As I wrote in the docs, right now we are expecting the server to report the same version even if it has been forced. If we decide to allow the server version to be forced, we should put that in the db module rather than trying to figure out where it would need to be ignored.

Reply:
Oh, I see what you meant now.

> If we decide to allow the server version to be forced, we should put that in the db module rather than trying to figure out where it would need to be ignored.

Agreed. I'll remove the `&& !cfgOptionTest(cfgOptPgVersion)) ` part.

> As I wrote in the docs, right now we are expecting the server to report the same version even if it has been forced.

Hm, I might have missed it but that's not clear to me from the help content :thinking: 
`The specified <postgres/> version will be used instead of the version automatically detected` couldn't mean from a "beginner" point of view that we're already forcing the db part too?